apps/managedidentity/managedidentity.go

```diff
@@ -323,9 +364,55 @@ func authResultFromToken(authParams authority.AuthParams, token accesstokens.Tok
 	return ar, err
 }
 
-func (client Client) getTokenForRequest(req *http.Request) (accesstokens.TokenResponse, error) {
-	var r accesstokens.TokenResponse
-	resp, err := client.httpClient.Do(req)
+// Contains checks if the element is present in the list.
+func contains[T comparable](list []T, element T) bool {
+	for _, v := range list {
+		if v == element {
+			return true
+		}
+	}
+	return false
+}
+
+// retry performs an HTTP request with retries based on the provided options.
+func (c Client) retry(maxRetries int, req *http.Request) (*http.Response, error) {
+	var resp *http.Response
+	var err error
+	for attempt := 0; attempt < maxRetries; attempt++ {
+		tryCtx, tryCancel := context.WithTimeout(req.Context(), time.Second*15)
+		defer tryCancel()
+		if resp != nil && resp.Body != nil {
+			io.Copy(io.Discard, resp.Body)
```

Comment:
linters warn about implicitly ignoring errors
```suggestion
			_, _ = io.Copy(io.Discard, resp.Body)
```